### ISSUE TITLE
gog.com

### ISSUE THREAD

**Issue by akmLaVx** [AdguardTeam/AdguardFilters#84828]
***Description***:
* **Current behaviour**: 

Downloading anything from gog.com might sporadically result in downloaded files being corrupted.

* **Workaround**: 

Adding gog.com to HTTPS exclusions list seems to work.

***Steps to reproduce the problem***:

1. Go to gog.com and try to download anything from the site (owned game, anything else)
2. With some probability, download will end up corrupted

***System configuration***

**Filters:**

REPRO

Information                            | Value
---                                    | ---
Operating system:                      | Windows
Operating system version (Android/iOS) | Windows 10 Pro 21H1
Browser:                               | Edge
AdGuard version:                       | AdGuard 7.7 nightly 1
Filters enabled:                       | Base, Russian, Mobile Ads, Tracking Protection, Annoyances, Adblock Warning Removal List, NoCoin Filter List, Experimental
AdGuard DNS:                           | None
Stealth mode options (Windows only)    | Stealth Mode Disabled
Helpdesk ID (if exists):               | Z514708

[//]: # (This template is meant for missed ad/false positive reports, for other type of reports edit it accordingly)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)

**Comment by Alex-302** [comment]
@akmLaVx I can't reproduce.

**Comment by D0G371** [comment]
I'm the one who informed AdGuard support of this issue directly via email, I didn't want to create a GitHub account just for this. Seems though the email was not directed directly to developers, instead an issue was opened here which is fine but for some reason all the details I gave how to reproduce this issue were not included.

This issue affects only offline installer files from gog.com for games with more than one .bin file, I can download any game with only the one .exe or the .exe file and one .bin file with no issues. The problem arises with games that have more than those two files, if I do not have gog.com listed in the HTTPS filtering exclusion list every file after the first .bin file is corrupted starting with the second .bin file.

Example using the game Dishonored 2:
.exe (no issues, MD5 hash match)
-1.bin (no issues, MD5 hash match)
-2.bin (corrupt, installer will fail integrity check and MD5 hash does not match)
-3.bin (corrupt, installer will fail integrity check and MD5 hash does not match)
-4.bin (corrupt, installer will fail integrity check and MD5 hash does not match)
-5.bin (corrupt, installer will fail integrity check and MD5 hash does not match)
-6.bin (corrupt, installer will fail integrity check and MD5 hash does not match)
-7.bin (corrupt, installer will fail integrity check and MD5 hash does not match)

I can reproduce this issue with any game I own that has more than one .bin file. Once I add gog.com to exclusion list all files can be downloaded without corruption.

**Comment by Alex-302** [comment]
@D0G371 Hi. These files?(only one game with multifile installer which I can download)
![image](https://user-images.githubusercontent.com/8361299/121510998-e1a5b700-c9f0-11eb-8669-90b318c2d988.png)

**Comment by Alex-302** [comment]
Downloaded without and with stealth mode, but game installed fine.
Downloaded files:
```
patch_full_throttle_remastered_1.1.879806_(11348)_to_1.1.891868_(14024).exe
setup_full_throttle_remastered_1.1.891868_(14024).exe
setup_full_throttle_remastered_1.1.891868_(14024)-1.bin
setup_full_throttle_remastered_1.1.891868_(14024)-2.bin
```

**Comment by Alex-302** [comment]
Please test when all filters, stealth mode and extensions module is disabled.
Probably some rare problem with filtering.

Before checking please enable debug level logging
https://kb.adguard.com/en/windows/solving-problems/adguard-logs
and export them(with time stamp of issue), if issue will be reproduced.

**Comment by D0G371** [comment]
@Alex-302 I went ahead and tried Full Throttle Remastered since I too own it, removed gog.com from the exclusion list and had no issues with it, no corruption. I wondered did the smaller size of that second .bin file (that part 3 of 3) had something to do with it and just to make sure I tried to download couple other files that most definitely has previously corrupted during download without exception, to my surprise all of sudden now they too were fine. I tried numerous times previously and every time the files corrupted without gog.com excluded and were fine when gog.com was excluded.

Only thing that has changed during the week has been windows updates and automatic AdGuard updates. I have no idea did they somehow fix the root cause, what ever that might be, but the issue seems to be fixed now.

It's starting to look like this was some odd isolated issue on my end. I'll keep my eye on and will perfom a full clean windows installation if the issue returns and see does that fix it. For now I think this issue can be closed, since I've seen nobody else with this problem.
